navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/NavigationEngine.java

```diff
@@ -94,7 +93,9 @@ void onLocationChanged(DirectionsRoute directionsRoute, Location location) {
     for (Milestone milestone : milestones) {
       if (milestone.isOccurring(previousRouteProgress, routeProgress)) {
         for (MilestoneEventListener listener : milestoneEventListeners) {
-          listener.onMilestoneEvent(routeProgress, INSTRUCTION_STRING, milestone.getIdentifier());
+          // Create a new DefaultInstruction based on the current RouteProgress and Milestone identifier
+          DefaultInstruction instruction = new DefaultInstruction(routeProgress, milestone.getIdentifier());
```

Comment:
Should be the generic `Instruction` class like so: `Instruction instruction = new DefaultInstruction(`. This way in the future, if a dev wants to implement their own instruction engine, they can (though we currently don't offer a way to pass in custom implementations)